class
❪"utilities"❫
❪"systems"❫
⚯
⚞⚟
◀public: static▶ void ☀Handle_Exceptions() ❰
    try{throw;}
    catch(std::string const& e                      ){Report_Error_Mesage_With_No_Extra_Data  (e);}
    catch(const char* e                             ){Report_Error_Mesage_With_No_Extra_Data  (e);}
    catch(...)                                       {Report_Unknown_Error                    ( );}
❱

◀private: static▶ void ☀Report_Error_Mesage_With_No_Extra_Data(std::string const& e)❰
    Red_Error_Message(e);
❱

◀private: static▶ void ☀Report_Unknown_Error()❰
    Red_Error_Message("Unknown exception thrown. This should not happen.\nYou found a bug in the software.\nPlease report this.");
❱
